packages/tpetra/core/src/Tpetra_CrsGraph_decl.hpp

```diff
@@ -2126,12 +2129,24 @@ namespace Tpetra {
     /// (global) graph is fill complete.
     local_graph_type getLocalGraph () const;
 
+
   protected:
+
+
     void fillLocalGraph (const Teuchos::RCP<Teuchos::ParameterList>& params);
 
     //! Throw an exception if the internal state is not consistent.
     void checkInternalState () const;
 
+    /// \brief Swaps the data from *this with the data and maps from graph.
+    ///
+    /// \param graph [in/out] a crsGraph
+    void swap(CrsGraph<LocalOrdinal, GlobalOrdinal, Node> & graph);
+
+    // Friend the tester for CrsGraph::swap
+    template <class LO, class GO, class N> friend class Tpetra::crsGraph_Swap_Tester;
```

Comment:
Question to the experts about syntax:
should this be
`friend class Tpetra::crsGraph_Swap_Tester<LocalOrdinal, GlobalOrdinal, Node>;`
(without the template specification at the beginning of the line)?

Reply:
Fixed.